AppFramework/Core/GREYElementInteraction.m

```diff
@@ -806,7 +806,13 @@ - (NSError *)grey_errorToReturnForInteractionError:(GREYError *)interactionError
   NSString *hierarchy = [self grey_unifyAndExtractHierarchyFromError:interactionError];
 
   // Add information such as element matcher and any nested error info.
+  // Copy over the matcher details from the error info dictionary.
   NSMutableDictionary<NSString *, id> *userInfo = [[NSMutableDictionary alloc] init];
+  if ([interactionError isKindOfClass:[GREYError class]]) {
```

Comment:
Can you explain more why this is needed?

Reply:
In case a user or a test case passes in an NSError ** into an EarlGrey method. Without this check, the error handling tests were crashing.

Reply:
Could you share the details of the failing test? This means that we have an NSError being passed into GREYElementInteraction which should not happen as the TestLib GREYElementInteractionProxy creates a GREYError to be sent.

Reply:
Several tests would fail without this check. For example, testCustomNSErrorInAction in ErrorHandlingTest